src/ui/components/inputs/DateInput.tsx

```diff
@@ -58,133 +72,141 @@ export const FULL_DATE_VALIDATOR = {
+    const parts = value.split('/')
+    const dateStr = [parts[2], parts[1], parts[0]].join('-')
+    if (validateDate(dateStr, 'boolean')) return new Date(dateStr)
+    return null
+  }, [value])
 
-  const dayNb = Number(day)
-  const monthNb = Number(month)
-  const yearNb = Number(year)
-  const date = new Date(yearNb, monthNb - 1, dayNb)
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      if (Platform.OS !== 'web') {
+        if (inputRef.current) inputRef.current.focus()
+      }
+    }, 500)
+    return () => clearTimeout(timeout)
+  }, [])
+
+  const isValidDate = date instanceof Date && !isNaN(date.getTime())
+  const dateValidation: DateValidation = useMemo(() => {
+    const nextDateValidation = {
+      isComplete: false,
+      isValid: false,
+      isDateAboveMin: minDate ? false : true,
+      isDateBelowMax: maxDate ? false : true,
+    }
 
-  const isDayValid = DAY_VALIDATOR.isValid(dayNb) && DAY_VALIDATOR.hasRightLength(day)
-  const isMonthValid = MONTH_VALIDATOR.isValid(monthNb) && MONTH_VALIDATOR.hasRightLength(month)
-  const isYearValid = YEAR_VALIDATOR.isValid(yearNb) && YEAR_VALIDATOR.hasRightLength(year)
-  const doesDateExists = FULL_DATE_VALIDATOR.isValid(date, yearNb, monthNb, dayNb)
+    if (isValidDate && date && minDate) {
+      nextDateValidation.isDateAboveMin = date >= minDate
+    }
+    if (isValidDate && date && maxDate) {
+      nextDateValidation.isDateBelowMax = date <= maxDate
+    }
 
-  if (doesDateExists && props.minDate) {
-    dateValidation.isDateAboveMin = date >= props.minDate
-  }
-  if (doesDateExists && props.maxDate) {
-    dateValidation.isDateBelowMax = date <= props.maxDate
-  }
+    const [day, month, year] = value.split('/')
+    nextDateValidation.isComplete =
+      DAY_VALIDATOR.hasRightLength(day ?? '') &&
+      MONTH_VALIDATOR.hasRightLength(month ?? '') &&
+      YEAR_VALIDATOR.hasRightLength(year ?? '')
+
+    nextDateValidation.isValid =
+      isValidDate &&
+      nextDateValidation.isComplete &&
+      nextDateValidation.isDateAboveMin &&
+      nextDateValidation.isDateBelowMax
 
-  dateValidation.isComplete =
-    DAY_VALIDATOR.hasRightLength(day) &&
-    MONTH_VALIDATOR.hasRightLength(month) &&
-    YEAR_VALIDATOR.hasRightLength(year)
-  dateValidation.isValid =
-    doesDateExists &&
-    dateValidation.isComplete &&
-    dateValidation.isDateAboveMin &&
-    dateValidation.isDateBelowMax
-
-  // notify parent effect
+    return nextDateValidation
+  }, [isValidDate, date, value])
   useEffect(() => {
-    props.onChangeValue?.(doesDateExists ? date : null, dateValidation)
-  }, [day, month, year])
+    props.onChangeValue?.(isValidDate && date ? date : null, dateValidation)
+  }, [dateValidation, isValidDate, date])
 
   useImperativeHandle(forwardedRef, () => ({
     clearFocuses() {
-      dayInputRef?.current?.blur()
-      monthInputRef?.current?.blur()
-      yearInputRef?.current?.blur()
+      inputRef.current?.blur?.()
```

Comment:
il est utile le ?. à la fin ?

Reply:
oui car si current est undefined, j'aurai une erreur "cannot execute blur() of undefined"